subprojects/kotlin-dsl/src/integTest/kotlin/org/gradle/kotlin/dsl/integration/BuildScriptCompileAvoidanceIntegrationTest.kt

```diff
@@ -0,0 +1,498 @@
+package org.gradle.kotlin.dsl.integration
+
+import org.gradle.integtests.fixtures.BuildOperationsFixture
+import org.gradle.integtests.fixtures.ToBeFixedForConfigurationCache
+import org.gradle.kotlin.dsl.fixtures.AbstractKotlinIntegrationTest
+import org.gradle.kotlin.dsl.provider.BUILDSCRIPT_COMPILE_AVOIDANCE_ENABLED
+import org.gradle.kotlin.dsl.provider.SCRIPT_CACHE_BASE_DIR_OVERRIDE_PROPERTY
+import org.gradle.test.fixtures.file.TestNameTestDirectoryProvider
+import org.hamcrest.CoreMatchers.containsString
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.Assert.assertTrue
+import org.junit.Assume.assumeTrue
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import java.util.regex.Pattern
+
+
+class BuildScriptCompileAvoidanceIntegrationTest : AbstractKotlinIntegrationTest() {
```

Comment:
Something I thought about today...

- I have a plugin that registers an extension with name `foo`. This plugin could be written in whatever language or come from a precompiled script plugin.
- I have a build script that applies the plugin and configures the extension by name. (e.g., `foo { /* configure */ }`)
- I run the build once and then change the plugin to register the extension with the name `bar`.
- I rerun the build, is the buildscript rebuilt?

I *think* this should cause us to generate new accessors and that means there's an ABI change. Is that what happens?

Reply:
Confirmed that this is what happens with a test.